### PR TITLE
PR: add pyalpm.vercmp as alternative to parse_version

### PR DESCRIPTION
**Pull request by farseerfc**
`pkg_resources.parse_version` has the different result from
`pyalpm.vercmp`. This change to github repos allows to use
`pyalpm.vercmp` as sort key.

This is just a PR for review comments. If this PR is accepted,
I will further implement this option in other repos and core.

### CODE REVIEW

File: nvchecker/source/github.py [hunk 1]
```diff
@@ -5,6 +5,29 @@
 from pkg_resources import parse_version
 from tornado.httpclient import AsyncHTTPClient, HTTPRequest
 
+try:
+  import pyalpm
+  def cmp_to_key(mycmp):
```
Comment (lilydjwg) [inline nvchecker/source/github.py]: Don't reinvent the wheel, just use `functools.cmp_to_key`.

File: nvchecker/source/github.py [hunk 2]
```diff
@@ -15,6 +38,7 @@ def get_version(name, conf, callback):
   use_latest_release = conf.getboolean('use_latest_release', False)
   use_max_tag = conf.getboolean('use_max_tag', False)
   ignored_tags = conf.get("ignored_tags", "").split()
+  sort_version_key = {"parse_version": parse_version, "vercmp": vercmp}[conf.get("sort_version_key", "parse_version")]
```
Comment (lilydjwg) [inline nvchecker/source/github.py]: I prefer to put the dict at global level.

File: nvchecker/source/github.py [hunk 3]
```diff
@@ -5,6 +5,29 @@
...
+        return mycmp(self.obj, other.obj) == 0
+      def __le__(self, other):
+        return mycmp(self.obj, other.obj) <= 0
+      def __ge__(self, other):
+        return mycmp(self.obj, other.obj) >= 0
+      def __ne__(self, other):
+        return mycmp(self.obj, other.obj) != 0
+  vercmp = cmp_to_key(pyalpm.vercmp)
+except ImportError:
+  vercmp = None
```
Comment (lilydjwg) [inline nvchecker/source/github.py]: If it's not available, we should give errors, or users may get unexpected results.

**Comment by farseerfc** [comment]
Changed to use functools.cmp_to_key.

I need some advise on:
 - Is there a way to share this code among different repo formats (github, gitlab, bitbuckts...) ?
 - How to behave like the original behavior when pyalpm is not available unless the config file specified vercmp?

**Comment by lilydjwg** [comment]
1. There isn't such a way yet. But you can add one to handle common options.
2. Set `vercmp` to `NotImplemented` instead of `None`, and when you get it, give out an error.

**Comment by farseerfc** [comment]
Implemented in most repo formats but not vcs. (I am still trying to figure out the callback chain in vcs)

File: nvchecker/lib/sortversion.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+logger = logging.getLogger(__name__)
+
+from pkg_resources import parse_version
+try:
+  import pyalpm
+  vercmp = cmp_to_key(pyalpm.vercmp)
+except ImportError:
+  def vercmp(k):
+    logger.error('Using vercmp but pyalpm can not be imported!')
+    return k
```
Comment (lilydjwg) [inline nvchecker/lib/sortversion.py]: 我们需要在比较版本的时候报错，你这样子会报错，但是并没有返回表示错误的结果（`None`）。另外报错信息里没有导致报错的软件名称。

**Comment by lilydjwg** [comment]
另外 lib 目录是放来自 winterpy 的模块的。

**Comment by farseerfc** [comment]
現在直接 raise 了 NotImplemented ，這樣可以麼？

**Comment by lilydjwg** [comment]
呃，你不能 raise NotImplemented 的，只能 raise NotImplementedError。

**Comment by farseerfc** [comment]
唔，我錯了，然後改掉了

**Comment by lilydjwg** [comment]
I've squashed all the commits into one :-)

**Pull request closed without merging**